pkg/linode-bs/controllerserver.go

```diff
@@ -349,15 +349,16 @@ func (linodeCS *LinodeControllerServer) ListVolumes(ctx context.Context, req *cs
 
 	listOpts := linodego.NewListOptions(0, "")
 	if req.GetMaxEntries() > 0 {
-		if startingToken == "" {
-			listOpts.Page = 1
-		} else {
-			startingPage, errParse := strconv.ParseInt(startingToken, 10, 64)
-			if errParse != nil {
-				return nil, status.Error(codes.Aborted, fmt.Sprintf("Invalid starting token %v", startingToken))
-			}
-			listOpts.Page = int(startingPage)
+		listOpts.PageSize = int(req.GetMaxEntries())
```

Comment:
On that note, the minimum is 25. I looked around trying to figure out what actually sets MaxEntries. Where does that value come from? Regardless, I think it may be best to leave that for the api to complain about, rather than have a magic 500.

Reply:
CSI services that make calls to this service set MaxEntires: for example, the `external-attacher`:

https://github.com/kubernetes-csi/external-attacher/blob/2d2ab15c96d6b22e9fb7c290b22bafa8d2cee4e1/pkg/attacher/lister.go#L46

In this case MaxEntires has the zero-value of 0

Reply:
Leaving the error cases up to the Linode API doesn't seem ideal to me, but there are cases that would require a lot of extra logic. For example, if MaxEntires was 10, then we would need "overlapping" API requests between iterations of this function.